src/toTransform.js

```diff
@@ -80,23 +104,28 @@
 
     pathElement.setAttribute('d', offsetPath.input);
 
-    var totalPathLength = pathElement.getTotalLength();
-    var offsetDistanceLength = getOffsetDistanceLength(offsetDistance, totalPathLength);
+    var currentOffsetDistance = getPathStringOffsetDistance(offsetPath, pathElement, offsetDistance);
 
-    if (closedLoop) {
-      if (offsetDistanceLength < 0) {
-        offsetDistanceLength = (offsetDistanceLength % totalPathLength) + totalPathLength;
-      } else {
-        offsetDistanceLength = offsetDistanceLength % totalPathLength;
-      }
-    } else if (offsetDistanceLength > totalPathLength) {
-      offsetDistanceLength = totalPathLength;
+    var epsilon = 0.001;
+    var nextOffsetDistanceValue = offsetDistance.value + epsilon;
+    if (!closedLoop && (currentOffsetDistance + epsilon) > totalPathLength) {
```

Comment:
Comparing against totalPathLength is only correct for px.
You would need to compare against 100 for %

Add a test for distance 100% with an open path.